### ISSUE TITLE
Please replace these firmwares with new ones

### ISSUE THREAD

**Issue by sq5bpf** [Tunas1337/UV-K5-Modded-Firmwares#58]
Please replace these firmwares with new ones which don't rely on band 1 being patched for full range. I think it's the way uvmod_kitchen patches work now.

The reason for this is that different firmwares behave differently when working out of band or in AM (some need band 1 to be set for AM regardless of frequency for example).  And i need a consistent way for chirp to set bands.

**Comment by lohtse** [comment]
just use a stock one and mod too suit your needs. Simple as that!!

**Comment by sq5bpf** [comment]
I know i can take a firmware and mod it. However the reason is that many people take firmwares from this repository, and i would like chirp to work with all of them, without introducing a lot of options that will just confuse everyone.

**Comment by lohtse** [comment]
chirp works with all you just need to go to the driver tab and tick the box....


And besides if you are putting a modded firmware on a device then you should have some un derstanding on what it does etc ...

**Comment by sq5bpf** [comment]
I'm the chirp driver author, and i know it doesn't work with all moded firmware.
I can introduce various hacks in the driver to take into account different hacked firmware behaviors, but this will just confuse the hell out of everyone, and there is no way to reliably autodetect that a particular hacked version is being used.

It would be much better to just use the enable AM on any band patch, and to expand the bands. This will be better for the users, and save me a lot of time.

**Comment by lohtse** [comment]
> I'm the chirp driver author, and i know it doesn't work with all moded firmware. I can introduce various hacks in the driver to take into account different hacked firmware behaviors, but this will just confuse the hell out of everyone, and there is no way to reliably autodetect that a particular hacked version is being used.
> 
> It would be much better to just use the enable AM on any band patch, and to expand the bands. This will be better for the users, and save me a lot of time.

If you are indeedd the author then you already know that the issue will happen reqardles  and that will be due to the same mods being added by the user..    In other words regardes of your request the issue will persist no matter what..  As to the Chirp the only issue I have had on any mod I have used is that it sometimes kicks up the out of range for some channels, most notabily those around 70-88mhz   to solve that I input using CPS...

**Comment by sq5bpf** [comment]
The very reason i want the firmwares to be replaced (and not new ones added)  is that i want them to behave more consistently for users. And these firmwares can have the same or better functionality.

This won't solve 100% of the problems, but will make it easier for people who often have a bit less technical ability and use ready-made firmwares.

But i think you can close this issue if you don't want to replace the firmwares. I will implement some option for users to set this from the driver side, it will be very confusing for most people, but maybe it will work in more cases.

**Comment by lohtse** [comment]
> The very reason i want the firmwares to be replaced (and not new ones added) is that i want them to behave more consistently for users. And these firmwares can have the same or better functionality.
> 
> This won't solve 100% of the problems, but will make it easier for people who often have a bit less technical ability and use ready-made firmwares.
> 
> But i think you can close this issue if you don't want to replace the firmwares. I will implement some option for users to set this from the driver side, it will be very confusing for most people, but maybe it will work in more cases.

will you be able to solve the out of range issue I mentioned?

**Comment by sq5bpf** [comment]
Yes, i will just make the ranges configurable. Not nice, and will confuse the hell out of most people, but maybe this is the right thing to do.

**Comment by Tunas1337** [comment]
I'll try to replace "all freqs in Band 2" with a patch that just enables AM everywhere. That may take a while though.On Aug 29, 2023 07:01, sq5bpf ***@***.***> wrote:
Yes, i will just make the ranges configurable. Not nice, and will confuse the hell out of most people, but maybe this is the right thing to do.
